### PR TITLE
Feat:Add userProfile entity

### PR DESCRIPTION
**Pull request by TeukuRyanIskandar**
image

- Added create, update, delete date on user entity
- Create userProfile entity and made one-to-one relation with user
- Error could be due to not migrating userProfile (not sure how to do that)

### CODE REVIEW

File: src/app/models/entities/userProfile.entity.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { User } from "@repositories/user.repository";
+import {
+  Column,
+  Entity,
+  OneToOne,
+  PrimaryGeneratedColumn,
+} from "typeorm";
+
+@Entity({ name: "user_profile" })
```
Comment (RKPinata) [inline src/app/models/entities/userProfile.entity.ts]: [Suggestion] Make the table name plural? 
```suggestion
@Entity({ name: "user_profiles" })
```

File: src/app/repositories/user.repository.ts [hunk 2]
```diff
@@ -3,4 +3,4 @@ import { AppDataSource } from "@root/ormconfig";
 
 const userRepository = AppDataSource.getRepository(User);
 
-export { userRepository };
+export { User, userRepository };
```
Comment (RKPinata) [inline src/app/repositories/user.repository.ts]: [Q] Why export the `User` entity?

Comment (RKPinata) [reply src/app/repositories/user.repository.ts]: [Request for change] Remove `User` entity export
```suggestion
export { userRepository };
```

File: src/app/models/schema/user.schema.ts [hunk 3]
```diff
@@ -25,4 +25,10 @@ export const createUserSchema: z.ZodSchema<CreateUserDto> =
         message:
           "Password must be at least 6 characters long",
       }),
+
+    createdAt: z.date(),
+
+    updatedAt: z.date(),
+
+    deletedAt: z.date(),
```
Comment (RKPinata) [inline src/app/models/schema/user.schema.ts]: [Q] This `createUserSchema` schema is for  validating the request body/payload `CreateUserDTO`.
 
The frontend wouldn't need to pass these fields. Why add it to the schema? These are generated fields right?

File: src/app/repositories/userProfile.repository.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { UserProfile } from "@models/entities";
+import { AppDataSource } from "@root/ormconfig";
+
+const userProfileRepository =
+  AppDataSource.getRepository(UserProfile);
+
+export { UserProfile, userProfileRepository };
```
Comment (RKPinata) [inline src/app/repositories/userProfile.repository.ts]: [R] Same concerns, remove `UserProfile` from export
```suggestion
export { userProfileRepository };
```

**Review by RKPinata — COMMENTED** [review]
Exporting the `User` and `UserProfile` entities within the repositories are causing the error.

**Comment by TeukuRyanIskandar** [comment]
Input:
http://localhost:4000/user/create
{
	"username": "teukurian98",
	"email": "email1@gmail.com",
	"password": "12345678"
}

Terminal:
error: [UserService][createUser] No metadata for "User" was found.
info: {"method":"POST","url":"/user/create","status":"500","responseTime":"62.938"}

insomnia:
{
	"code": 500,
	"data": null,
	"message": "Error creating user"
}

**Comment by TeukuRyanIskandar** [comment]
Changed
entities: ["src/app/models/*.entity{.ts,.js}"] --->   entities: ["src/app/models/entities/*.entity{.ts,.js}"] at ormconfig

query: START TRANSACTION
query: INSERT INTO "users"("username", "email", "password", "createdAt", "updatedAt", "deletedAt", "profileId") VALUES ($1, $2, $3, DEFAULT, DEFAULT, DEFAULT, DEFAULT) RETURNING "id", "createdAt", "updatedAt", "deletedAt" -- PARAMETERS: ["teukurian98","email1@gmail.com","$2b$10$txLHksVAP3yjvuunKugeX.Y6qxMCkvDEJYPdBnTjfu8wl/YKgQBm6"]
query failed: INSERT INTO "users"("username", "email", "password", "createdAt", "updatedAt", "deletedAt", "profileId") VALUES ($1, $2, $3, DEFAULT, DEFAULT, DEFAULT, DEFAULT) RETURNING "id", "createdAt", "updatedAt", "deletedAt" -- PARAMETERS: ["teukurian98","email1@gmail.com","$2b$10$txLHksVAP3yjvuunKugeX.Y6qxMCkvDEJYPdBnTjfu8wl/YKgQBm6"]
error: error: column "createdAt" of relation "users" does not exist
query: ROLLBACK
error: [UserService][createUser] column "createdAt" of relation "users" does not exist
info: {"method":"POST","url":"/user/create","status":"500","responseTime":"114.016"}

**Pull request closed without merging**